oap-server/server-storage-plugin/storage-jdbc-hikaricp-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/jdbc/h2/dao/H2BatchDAO.java

```diff
@@ -22,61 +22,85 @@
     }
 
     @Override
     public void flush(List<PrepareRequest> prepareRequests) {
         if (CollectionUtils.isEmpty(prepareRequests)) {
             return;
         }
-
         if (log.isDebugEnabled()) {
-            log.debug("batch sql statements execute, data size: {}", prepareRequests.size());
+            log.debug("to execute sql statements execute, data size: {}, maxBatchSqlSize: {}", prepareRequests.size(), maxBatchSqlSize);
+        }
+        executeSql(prepareRequests, maxBatchSqlSize);
+        if (log.isDebugEnabled()) {
+            log.debug("execute sql statements done, data size: {}, maxBatchSqlSize: {}", prepareRequests.size(), maxBatchSqlSize);
         }
+    }
 
+    private void executeSql(List<PrepareRequest> prepareRequests, int maxBatchSqlSize) {
+        if (log.isDebugEnabled()) {
+            log.debug("execute sql batch. data size:{}", prepareRequests.size());
+        }
+        Map<String, List<PrepareRequest>> batchRequestMap = new HashMap<>();
+        for (PrepareRequest prepareRequest : prepareRequests) {
+            SQLExecutor sqlExecutor = (SQLExecutor) prepareRequest;
+            if (batchRequestMap.containsKey(sqlExecutor.getSql())) {
+                batchRequestMap.get(sqlExecutor.getSql()).add(prepareRequest);
+            } else {
+                List<PrepareRequest> prepareRequestList = new ArrayList<>();
+                prepareRequestList.add(sqlExecutor);
+                batchRequestMap.put(sqlExecutor.getSql(), prepareRequestList);
+            }
+        }
         try (Connection connection = h2Client.getConnection()) {
-            for (PrepareRequest prepareRequest : prepareRequests) {
-                try {
-                    SQLExecutor sqlExecutor = (SQLExecutor) prepareRequest;
-                    sqlExecutor.invoke(connection);
-                } catch (SQLException e) {
-                    // Just avoid one execution failure makes the rest of batch failure.
-                    log.error(e.getMessage(), e);
+            try {
+                Set<Map.Entry<String, List<PrepareRequest>>> entrySet = batchRequestMap.entrySet();
+                Iterator<Map.Entry<String, List<PrepareRequest>>> iterator = entrySet.iterator();
+                while (iterator.hasNext()) {
+                    Map.Entry<String, List<PrepareRequest>> next = iterator.next();
+                    BatchSQLExecutor batchSQLExecutor = new BatchSQLExecutor(next.getKey(), next.getValue());
+                    batchSQLExecutor.invoke(connection, maxBatchSqlSize);
                 }
+            } catch (SQLException e) {
+                // Just avoid one execution failure makes the rest of batch failure.
+                log.error(e.getMessage(), e);
             }
+
         } catch (SQLException | JDBCClientException e) {
+            log.warn("execute sql failed, discard data size: {}", prepareRequests.size());
```

Comment:
```suggestion
            log.warn("execute sql failed, discard data size: {}", prepareRequests.size(), e);
```

And remove the next line `log.error(e.getMessage(), e);`

Reply:
ok done